config/hooks/pre-commit

```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+set -e
+
+REPO_ROOT_DIR="$(git rev-parse --show-toplevel)"
+
+files=$((git diff --cached --name-only --diff-filter=ACMR | grep -Ei "\.java$") || true)
+if [ ! -z "${files}" ]; then
```

Comment:
Simplify by using `-n` instead of `! -z`.

Caught by `shellcheck` linter.

Reply:
Simplify by  using `"$files"` instead of `"${files}"`. No need for curly braces here.